# Task 4: Optimization Model using Linear Programming

## Implementation using PuLP

In [3]:
# Task 4: Optimization Problem

# A factory produces two products: A and B

# Profit:
# A → ₹20 per unit
# B → ₹30 per unit

# Constraints:
# 1. Each unit of A requires 2 hours, B requires 3 hours
#    Total available time = 100 hours

# 2. Each unit of A requires 1 material, B requires 2 materials
#    Total material available = 80 units

# Objective:
# Maximize total profit

In [4]:
from pulp import LpMaximize, LpProblem, LpVariable

# Create the model
model = LpProblem(name="profit-maximization", sense=LpMaximize)

# Define variables
x = LpVariable(name="Product_A", lowBound=0)
y = LpVariable(name="Product_B", lowBound=0)

# Objective function
model += 20 * x + 30 * y

# Constraints
model += 2 * x + 3 * y <= 100
model += x + 2 * y <= 80

# Solve
model.solve()

# Output results
print("Product A:", x.value())
print("Product B:", y.value())
print("Maximum Profit:", model.objective.value())

Product A: 0.0
Product B: 33.333333
Maximum Profit: 999.9999900000001


## Results and Insights

The model suggests producing 0 units of Product A and approximately 33.33 units of Product B.

The maximum profit achieved is around ₹1000.

This shows that Product B is more profitable. Time is the limiting factor in this problem, while material is not fully utilized.

The optimization model suggests producing 0 units of Product A and approximately 33.33 units of Product B to achieve maximum profit.

The maximum profit obtained is around ₹1000.

This indicates that Product B is more profitable and efficiently uses the available time resource. Time is the limiting constraint in this problem, while material is not fully utilized.